src/materials/MeshMatcapMaterial.js

```diff
@@ -0,0 +1,129 @@
+import { TangentSpaceNormalMap } from '../constants.js';
+import { Material } from './Material.js';
+import { Vector2 } from '../math/Vector2.js';
+import { Color } from '../math/Color.js';
+
+/**
+ * @author WestLangley / http://github.com/WestLangley
+ *
+ * parameters = {
+ *  color: <hex>,
+ *  opacity: <float>,
+ *
+ *  matcap: new THREE.Texture( <Image> ),
+ *
+ *  map: new THREE.Texture( <Image> ),
+ *
+ *  bumpMap: new THREE.Texture( <Image> ),
+ *  bumpScale: <float>,
+ *
+ *  normalMap: new THREE.Texture( <Image> ),
+ *  normalMapType: THREE.TangentSpaceNormalMap,
+ *  normalScale: <Vector2>,
+ *
+ *  displacementMap: new THREE.Texture( <Image> ),
+ *  displacementScale: <float>,
+ *  displacementBias: <float>,
+ *
+ *  alphaMap: new THREE.Texture( <Image> ),
+ *
+ *  skinning: <bool>,
+ *  morphTargets: <bool>,
+ *  morphNormals: <bool>
+ * }
+ */
+
+function MeshMatcapMaterial( parameters ) {
+
+	Material.call( this );
+
+	this.defines = { 'MATCAP': '' };
+
+	this.type = 'MeshMatcapMaterial';
+
+	this.color = new Color( 0xffffff ); // diffuse
+
+	this.matcap = null;
+
+	this.map = null;
+
+	this.bumpMap = null;
+	this.bumpScale = 1;
+
+	this.normalMap = null;
+	this.normalMapType = TangentSpaceNormalMap;
+	this.normalScale = new Vector2( 1, 1 );
+
+	this.displacementMap = null;
+	this.displacementScale = 1;
+	this.displacementBias = 0;
+
+	this.alphaMap = null;
+
+	this.skinning = false;
+	this.morphTargets = false;
+	this.morphNormals = false;
+
+	this.lights = false;
+
+	this.setValues( parameters );
+
+	// a matcap is required
+
+	if ( this.matcap === null ) {
+
+		var canvas = document.createElement( 'canvas' );
```

Comment:
@mrdoob Is referencing `document` here going to be a problem?

Is there another way we can have a default matcap?

This is actually not necessary. If the matcap is null, the model will render black, I guess. But for drag and drop support, ensuring the matcap was a valid texture to begin with makes things easy.

Reply:
This is how it's done in `WebGLTextures`:
https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLTextures.js#L31

Reply:
You can probably create a `DataTexture` instead though.

Reply:
>document.createElementNS

That presupposes a document, too. Is that OK? I'm not sure of the advantage of `createElementNS` here...

>You can probably create a DataTexture instead though.

I tried a `DataTexture` first, but there were errors when the user replaced the default matcap with a `Texture`, or when drag-and-drop was used. It is possible I did something wrong, of course...

Reply:
> I tried a DataTexture first, but there were errors when the user replaced the default matcap with a Texture, or when drag-and-drop was used. It is possible I did something wrong, of course...

Oh, sounds like a bug. I'll have to investigate.

Reply:
>I'll have to investigate.

Sweet. I appreciate that!

Here is my original attempt:
```javascript
if ( this.matcap === null ) {

	this.matcap = new THREE.DataTexture( new Uint8Array( [ 255, 255, 255 ] ), 1, 1, THREE.RGBFormat, THREE.UnsignedByteType ); // white

	this.matcap.needsUpdate = true;

}
```

Reply:
>Oh, sounds like a bug. I'll have to investigate.

/bump

/ping @Mugen87 Should it be this, instead? Or doesn't it matter?

```
var canvas = document.createElementNS( 'http://www.w3.org/1999/xhtml', 'canvas' );
```

Reply:
Yes, that's correct. Otherwise we assume a HTML document context. Explicitly defining the HTML namespace is more robust.

Reply:
>> Oh, sounds like a bug. I'll have to investigate.
>
> /bump

#15313